src/main/kotlin/com/example/Employee/controllers/HolidaysController.kt

```diff
@@ -19,23 +18,23 @@ import java.util.*
 public class HolidaysController (private var holidaysService: HolidaysService){
 
     @PostMapping
-    fun createHoliday(@RequestBody holiday: Holidays):ResponseEntity<Holidays> {
+    fun createHoliday(@RequestBody holiday: Holiday):ResponseEntity<Holiday> {
         holidaysService.createHoliday(holiday)
         return ResponseEntity.ok(holiday)
     }
 
     @GetMapping
-    fun getHolidays(): MutableIterable<Holidays> {
+    fun getHolidays(): MutableIterable<Holiday> {
         return holidaysService.getHolidays()
     }
 
     @GetMapping("/{id}")
-    fun getHoliday(@PathVariable id: Int): Optional<Holidays> {
+    fun getHoliday(@PathVariable id: Int): Optional<Holiday> {
```

Comment:
If the `Optional` object is empty, your API response must be 404. So, you can't return the Optional<Holiday> object.